### ISSUE TITLE
No longjmp() during early init

### ISSUE THREAD

**Issue by tdz** [picotm/picotm#139]
If the early initialization of the transaction structure fails, there's no longjmp() target yet. Either don't do longjmp() here or make sure that setjmp() has been called before.

**Comment by tdz** [comment]
The stack trace is

```
Thread 3 "picotm-perf" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff6636700 (LWP 12720)]
__longjmp () at ../sysdeps/x86_64/__longjmp.S:67
67              jmpq *%rdx
(gdb) bt
#0  __longjmp () at ../sysdeps/x86_64/__longjmp.S:67
#1  0x6e5c5d72bd95bfb7 in ?? ()
Backtrace stopped: Cannot access memory at address 0x6e5c5d72bd95bfb7
```

This happened after calling picotm_recover_from_error() during the early init.

**Comment by tdz** [comment]
Some more infos: `__picotm_get_tx()` initializes the tx structure. If this operation fails, `longjmp()` gets invoked for error handling. The jump buffer is part of the data structures initialized by `__picotm_get_tx()`, so the `longjmp()` goes nowhere.

**Comment by tdz** [comment]
Fixed by PR #170
